src/inspect_ai/log/_file.py

```diff
@@ -135,6 +136,8 @@ def write_eval_log(
        location (str | FileInfo): Location to write log to.
        format (Literal["eval", "json", "auto"]): Write to format
           (defaults to 'auto' based on `log_file` extension)
+       if_match_etag (str | None): ETag for conditional write. If provided
```

Comment:
We should note in the doc comments that `WriteConflictError` is raised if the eTag doesn't match.